RecoEgamma/EgammaIsolationAlgos/interface/EgammaL1TkIsolation.h

```diff
@@ -0,0 +1,58 @@
+#ifndef RECOEGAMMA_EGAMMAISOLATIONALGOS_EGAMMAL1TKISOLATION_H
+#define RECOEGAMMA_EGAMMAISOLATIONALGOS_EGAMMAL1TKISOLATION_H
+
+#include "DataFormats/L1TrackTrigger/interface/L1Track.h"
+#include "DataFormats/TrackReco/interface/TrackBase.h"
+#include "DataFormats/TrackReco/interface/TrackFwd.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+
+//author S. Harper (RAL/CERN)
+//based on the work of Swagata Mukherjee and Giulia Sorrentino
+
+class EgammaL1TkIsolation {
+public:
+private:
+  struct TrkCuts {
+    float minPt;
+    float minDR2;
+    float maxDR2;
+    float minDEta;
+    float maxDZ;
+    explicit TrkCuts(const edm::ParameterSet& para);
+    static edm::ParameterSetDescription pSetDescript();
```

Comment:
It might be nicer to call this `fillPSetDescription()`, following the naming scheme from https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideConfigurationValidationAndHelp#A_Plugin_Module_Using_Another_He (also for the instance below). Even better if it follows the pattern with `edm::ParameterSetDescription& iDesc` provided as an argument rather than a return value.

Reply:
so this makes it cumbersome to use later on, eg here: https://github.com/cms-sw/cmssw/blob/68e08249ad2976a3675000120848a07423b34957/RecoEgamma/EgammaHLTProducers/plugins/EgammaHLTEleL1TrackIsolProducer.cc#L64

However hypothetically (though unlikely) somebody may wish to have this functionality and so I have provided both interfaces.

Reply:
That line would just need to change to:
```
EgammaL1TkIsolation::fillPSetDescription(desc);
```
(and then `fillPSetDescription()` would call `desc.add(...)`)

Reply:
still two lines and less clear than before? I dont see the problem with makePSetDescription() given I also supply a fillPSetDescription()

Reply:
@makortel please clarify how strongly the recommended pattern should be enforced

Reply:
I think the `makePSetDescription()` that returns a `edm::ParameterSetDescription` works fine here.

(for non-helper-plugin use cases I think the naming convention of `...PSetDescription` is more important than whether the `ParameterSetDescription` is passed as an out-argument or returned)